plugins/module_utils/utils.py

```diff
@@ -438,3 +441,35 @@ def wait_for_redfish_job_complete(redfish_obj, job_uri, job_wait=True, wait_time
         job_resp = redfish_obj.invoke_request("GET", job_uri, api_timeout=120)
         return job_resp, ""
     return job_resp, job_msg
+
+
+def get_dynamic_uri(idrac_obj, base_uri, search_label=''):
+    resp = idrac_obj.invoke_request(method='GET', uri=base_uri).json_data
+    if search_label and search_label in resp:
+        return resp[search_label]
+    return resp
```

Comment:
returning None would be better?

Reply:
I've handled both scenario 1. If I want to search some key in the response  2. If no key has been mentioned to look, it'll simply return whole resp, so user can process on that.... Any reason why None would be better ?

Reply:
how will the function know if he got what is searched for ? dict,list,str check?

Reply:
I can do one thing, if user has mentioned key and it's not found in list/dict then I can return None, else resp will return? are you okay with that ?